include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -0,0 +1,250 @@
+
+        if (sq_hamiltonian_beta.dimension() != dim) {
+            throw std::invalid_argument("USQHamiltonian::USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed): The dimensions of the alpha and beta Hamiltonian are incompatible");
+        }
+        
+        for (const auto& two_op : this->two_op_mixed) {
+            if (two_op.dimension() != dim) {
+                throw std::invalid_argument("USQHamiltonian::USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed): The dimensions of the mixed two electron operator are incompatible with the Hamiltonian");
+            }
+        }
+        
+        // Calculate the total two-electron operator
+        QCRankFourTensor<Scalar> total_two_op_par (dim);
+        total_two_op_par.setZero();
+        for (const auto& two_op : this->two_op_mixed) {
+            total_two_op_par += two_op.parameters().Eigen();
+        }
+        this->total_two_op_mixed = ScalarSQTwoElectronOperator<Scalar>({total_two_op_par});
+    }
+
+    /**
+     *  @param sq_hamiltonian_alpha      alpha hamiltonian
+     *  @param sq_hamiltonian_beta       beta hamiltonian
+     *  @param two_op_mixed              alpha & beta mixed operator
+     */
+    USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed) :
+        USQHamiltonian(sq_hamiltonian_alpha, sq_hamiltonian_beta, std::vector<ScalarSQTwoElectronOperator<Scalar>>({two_op_mixed}))
+    {}
+
+
+    /*
+     *  NAMED CONSTRUCTORS
+     */
+
+    /**
+     *  Construct the molecular Hamiltonian in a given single-particle basis
+     *
+     *  @param sp_basis     the single-particle basis in which the Hamiltonian should be expressed
+     *
+     *  @return a second-quantized molecular Hamiltonian. The molecular Hamiltonian has
+     *      - one-electron contributions:
+     *          - kinetic
+     *          - nuclear attraction
+     *      - two-electron contributions:
+     *          - Coulomb repulsion
+     *
+     *  Note that this named constructor is only available for real matrix representations
+     */
+    template <typename Z = Scalar>
+    static enable_if_t<std::is_same<Z, double>::value, USQHamiltonian<double>> Molecular(const SingleParticleBasis<Z, GTOShell>& sp_basis_alpha, const SingleParticleBasis<Z, GTOShell>& sp_basis_beta, const Molecule& molecule) {
+
+        const SQHamiltonian<Scalar> sq_hamiltonian_alpha = SQHamiltonian<double>::Molecular(sp_basis_alpha, molecule);
+        const SQHamiltonian<Scalar> sq_hamiltonian_beta = SQHamiltonian<double>::Molecular(sp_basis_beta, molecule);
+        const ScalarSQTwoElectronOperator<double> two_op_mixed = sq_hamiltonian_alpha.twoElectron();
```

Comment:
It seems to me that this constructor wants to initialize an unrestricted Hamiltonian that has a different expansion of the alpha spinors and beta spinors, which is a good thing! This would allow for larger flexibility to describe the alpha and beta spinors.

However, I don't really understand how `two_op_mixed` (`g^{aabb}`) is only calculated from the alpha Hamiltonian. Could you explain this in the code?

Reply:
This is under assumption that the initial basis will be same.  I will remove the functionality.